cpp/ql/src/semmle/code/cpp/dataflow/internal/DataFlowImpl.qll

```diff
@@ -1482,6 +1634,13 @@ abstract private class AccessPath extends TAccessPath {
     this = TConsCons(_, _, result)
   }
 
+  /**
+   * Holds if the length of this access path does not exceed the value
+   * of `flowThroughAccessPathLimit()`, if any.
+   */
+  pragma[noinline]
+  predicate isValidForFlowThrough() { not this.len() > flowThroughAccessPathLimit() }
```

Comment:
Come to think of it, perhaps the language-specific restriction should apply to all access paths, not just paths through calls. And perhaps even make it possible to override this restriction in a data flow configuration. What do you think @aschackmull ?

Reply:
Yes, let's just start out with a language-specific config-independent length bound, as that is the simplest.